nautobot/dcim/models/device_component_templates.py

```diff
@@ -387,3 +447,83 @@ def clean(self):
             raise ValidationError(
                 f'Subdevice role of device type ({self.device_type}) must be set to "parent" to allow device bays.'
             )
+
+
+@extras_features("custom_validators")
+class ModuleBayTemplate(BaseModel, ChangeLoggedModel, CustomFieldModel, RelationshipModel):
```

Comment:
Can this be a subclass of ModularComponentTemplate?

Reply:
Yeah, could subclass and override `name` to optional for this specific model, but that might be iffy if there's code assuming that all `ComponentTemplateModels` have a `name`?